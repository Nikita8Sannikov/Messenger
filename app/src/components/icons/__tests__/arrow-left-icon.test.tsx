import { fireEvent, render, screen } from "@testing-library/react";

import ArrowLeftIconComponent from "@components/icons/arrow-left";

describe("ArrowLeftIcon", () => {
	it("renders with default props", () => {
		render(<ArrowLeftIconComponent />);
		const icon = screen.getByTestId("arrow-left-icon");
		expect(icon).toBeInTheDocument();
		expect(icon).toHaveAttribute("width", "24px");
		expect(icon).toHaveAttribute("height", "24px");
	});

	it("renders with custom size as number", () => {
		render(<ArrowLeftIconComponent size={32} />);
		const icon = screen.getByTestId("arrow-left-icon");
		expect(icon).toHaveAttribute("width", "32px");
		expect(icon).toHaveAttribute("height", "32px");
	});

	it("renders with custom size as string", () => {
		render(<ArrowLeftIconComponent size="48px" />);
		const icon = screen.getByTestId("arrow-left-icon");
		expect(icon).toHaveAttribute("width", "48px");
		expect(icon).toHaveAttribute("height", "48px");
	});

	it("renders with custom className", () => {
		render(<ArrowLeftIconComponent className="custom-class" />);
		const icon = screen.getByTestId("arrow-left-icon");
		expect(icon).toHaveClass("custom-class");
	});

	it("calls onClick handler when clicked", () => {
		const handleClick = jest.fn();
		render(<ArrowLeftIconComponent onClick={handleClick} />);
		const icon = screen.getByTestId("arrow-left-icon");
		fireEvent.click(icon);
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	it("matches snapshot with default props", () => {
		const { container } = render(<ArrowLeftIconComponent />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with custom size as number", () => {
		const { container } = render(<ArrowLeftIconComponent size={32} />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with custom size as string", () => {
		const { container } = render(<ArrowLeftIconComponent size="48px" />);
		expect(container).toMatchSnapshot();
	});

	it("matches snapshot with custom className", () => {
		const { container } = render(<ArrowLeftIconComponent className="custom-class" />);
		expect(container).toMatchSnapshot();
	});
}); 