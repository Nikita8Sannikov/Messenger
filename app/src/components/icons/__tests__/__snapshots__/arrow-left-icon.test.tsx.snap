// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowLeftIcon matches snapshot with custom className 1`] = `
<div>
  <svg
    class="custom-class"
    data-testid="arrow-left-icon"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0"
    />
  </svg>
</div>
`;

exports[`ArrowLeftIcon matches snapshot with custom size as number 1`] = `
<div>
  <svg
    data-testid="arrow-left-icon"
    height="32px"
    viewBox="0 0 24 24"
    width="32px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0"
    />
  </svg>
</div>
`;

exports[`ArrowLeftIcon matches snapshot with custom size as string 1`] = `
<div>
  <svg
    data-testid="arrow-left-icon"
    height="48px"
    viewBox="0 0 24 24"
    width="48px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0"
    />
  </svg>
</div>
`;

exports[`ArrowLeftIcon matches snapshot with default props 1`] = `
<div>
  <svg
    data-testid="arrow-left-icon"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="0"
    />
  </svg>
</div>
`;
